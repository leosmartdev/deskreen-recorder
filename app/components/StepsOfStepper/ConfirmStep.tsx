import React, { useEffect, useState } from 'react';
import { remote } from 'electron';
import { Text } from '@blueprintjs/core';
import { Row, Col } from 'react-flexbox-grid';
import { useTranslation } from 'react-i18next';
import SharingSourcePreviewCard from '../SharingSourcePreviewCard';
import SharingSessionService from '../../features/SharingSessionService';
import DeviceInfoCallout from '../DeviceInfoCallout';
import SharingSession from '../../features/SharingSessionService/SharingSession';

const sharingSessionService = remote.getGlobal(
  'sharingSessionService'
) as SharingSessionService;

interface ConfirmStepProps {
  device: Device | null;
}

export default function ConfirmStep(props: ConfirmStepProps) {
  const { t } = useTranslation();
  const { device } = props;
  const [sharingSession, setSharingSession] = useState<
    SharingSession | undefined
  >();

  useEffect(() => {
    if (sharingSessionService.waitingForConnectionSharingSession !== null) {
      setSharingSession(
        sharingSessionService.waitingForConnectionSharingSession
      );
    }
  }, []);

  return (
    <div style={{ width: '80%', marginTop: '50px' }}>
      <Row style={{ marginBottom: '10px' }}>
        <Col xs={12} style={{ textAlign: 'center' }}>
          {/* eslint-disable-next-line react/no-unescaped-entities */}
          <Text>{t('Check if all is OK and click Confirm')}</Text>
        </Col>
      </Row>
      <Row middle="xs" center="xs">
        <Col xs={5}>
          <DeviceInfoCallout
            deviceType={device?.deviceType}
            deviceIP={device?.deviceIP}
            deviceOS={device?.deviceOS}
            sharingSessionID={device?.sharingSessionID}
            deviceBrowser={device?.deviceBrowser}
          />
        </Col>
        <Col xs={5}>
          <SharingSourcePreviewCard
            sharingSourceID={sharingSession?.desktopCapturerSourceID}
          />
        </Col>
      </Row>
    </div>
  );
}
