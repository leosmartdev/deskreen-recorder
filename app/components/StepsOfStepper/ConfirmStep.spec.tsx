import React from 'react';
import Enzyme, { mount } from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import ConfirmStep from './ConfirmStep';

Enzyme.configure({ adapter: new Adapter() });
jest.useFakeTimers();

it('should match exact snapshot', () => {
  const subject = mount(
    <>
      <Router>
        <ConfirmStep device={{} as Device} />
      </Router>
    </>
  );
  expect(EnzymeToJson(subject)).toMatchSnapshot();
});
