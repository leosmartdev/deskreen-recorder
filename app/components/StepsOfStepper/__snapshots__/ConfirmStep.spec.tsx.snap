// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match exact snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ConfirmStep
      device={Object {}}
    >
      <div
        style={
          Object {
            "marginTop": "50px",
            "width": "80%",
          }
        }
      >
        <Row
          style={
            Object {
              "marginBottom": "10px",
            }
          }
        >
          <div
            className="row"
            style={
              Object {
                "marginBottom": "10px",
              }
            }
          >
            <Col
              style={
                Object {
                  "textAlign": "center",
                }
              }
              xs={12}
            >
              <div
                className="col-xs-12"
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                <Blueprint3.Text
                  ellipsize={false}
                  tagName="div"
                >
                  <div
                    className=""
                  />
                </Blueprint3.Text>
              </div>
            </Col>
          </div>
        </Row>
        <Row
          center="xs"
          middle="xs"
        >
          <div
            className="row center-xs middle-xs"
          >
            <Col
              xs={5}
            >
              <div
                className="col-xs-5"
              >
                <DeviceInfoCallout>
                  <Component
                    style={
                      Object {
                        "margin": "0 auto",
                        "textAlign": "center",
                      }
                    }
                  >
                    <h4
                      className="bp3-heading"
                      style={
                        Object {
                          "margin": "0 auto",
                          "textAlign": "center",
                        }
                      }
                    >
                      :
                    </h4>
                  </Component>
                  <Blueprint3.Callout
                    id="device-info-callout"
                    style={
                      Object {
                        "borderRadius": "8px",
                      }
                    }
                  >
                    <div
                      className="bp3-callout"
                      id="device-info-callout"
                      style={
                        Object {
                          "borderRadius": "8px",
                        }
                      }
                    >
                      <Row
                        center="xs"
                      >
                        <div
                          className="row center-xs"
                        >
                          <Col
                            xs={12}
                          >
                            <div
                              className="col-xs-12"
                            >
                              <Blueprint3.Text
                                ellipsize={false}
                                tagName="div"
                              >
                                <div
                                  className=""
                                >
                                  :
                                   
                                  <span />
                                </div>
                              </Blueprint3.Text>
                              <Blueprint3.Tooltip
                                content={
                                  <React.Fragment>
                                    <Blueprint3.Text
                                      ellipsize={false}
                                      tagName="div"
                                    >
                                      
                                    </Blueprint3.Text>
                                    <span
                                      style={
                                        Object {
                                          "fontWeight": 900,
                                        }
                                      }
                                    >
                                      <Blueprint3.Text
                                        ellipsize={false}
                                        tagName="div"
                                      >
                                        
                                      </Blueprint3.Text>
                                    </span>
                                  </React.Fragment>
                                }
                                hoverCloseDelay={0}
                                hoverOpenDelay={100}
                                minimal={false}
                                position="top"
                                transitionDuration={100}
                              >
                                <Blueprint3.Popover
                                  autoFocus={false}
                                  boundary="scrollParent"
                                  canEscapeKeyClose={false}
                                  captureDismiss={false}
                                  content={
                                    <React.Fragment>
                                      <Blueprint3.Text
                                        ellipsize={false}
                                        tagName="div"
                                      >
                                        
                                      </Blueprint3.Text>
                                      <span
                                        style={
                                          Object {
                                            "fontWeight": 900,
                                          }
                                        }
                                      >
                                        <Blueprint3.Text
                                          ellipsize={false}
                                          tagName="div"
                                        >
                                          
                                        </Blueprint3.Text>
                                      </span>
                                    </React.Fragment>
                                  }
                                  defaultIsOpen={false}
                                  disabled={false}
                                  enforceFocus={false}
                                  fill={false}
                                  hasBackdrop={false}
                                  hoverCloseDelay={0}
                                  hoverOpenDelay={100}
                                  inheritDarkTheme={true}
                                  interactionKind="hover-target"
                                  lazy={true}
                                  minimal={false}
                                  modifiers={
                                    Object {
                                      "arrow": Object {
                                        "enabled": true,
                                      },
                                    }
                                  }
                                  openOnTargetFocus={true}
                                  popoverClassName="bp3-tooltip"
                                  position="top"
                                  targetTagName="span"
                                  transitionDuration={100}
                                  usePortal={true}
                                  wrapperTagName="span"
                                >
                                  <Manager>
                                    <span
                                      className="bp3-popover-wrapper"
                                    >
                                      <Reference
                                        innerRef={[Function]}
                                      >
                                        <InnerReference
                                          innerRef={[Function]}
                                          setReferenceNode={[Function]}
                                        >
                                          <Blueprint3.ResizeSensor
                                            onResize={[Function]}
                                          >
                                            <span
                                              className="bp3-popover-target"
                                              onBlur={[Function]}
                                              onFocus={[Function]}
                                              onMouseEnter={[Function]}
                                              onMouseLeave={[Function]}
                                            >
                                              <div
                                                className=""
                                                key=".0"
                                                style={
                                                  Object {
                                                    "backgroundColor": "#00f99273",
                                                    "borderRadius": "20px",
                                                    "fontWeight": 900,
                                                    "paddingLeft": "10px",
                                                    "paddingRight": "10px",
                                                  }
                                                }
                                                tabIndex={0}
                                              >
                                                <Blueprint3.Text
                                                  className="bp3-text-large"
                                                  ellipsize={false}
                                                  tagName="div"
                                                >
                                                  <div
                                                    className="bp3-text-large"
                                                  >
                                                    :
                                                     
                                                    <span
                                                      className="device-ip-span"
                                                    />
                                                  </div>
                                                </Blueprint3.Text>
                                              </div>
                                            </span>
                                          </Blueprint3.ResizeSensor>
                                        </InnerReference>
                                      </Reference>
                                      <Blueprint3.Overlay
                                        autoFocus={false}
                                        backdropClassName="bp3-popover-backdrop"
                                        backdropProps={Object {}}
                                        canEscapeKeyClose={false}
                                        canOutsideClickClose={false}
                                        enforceFocus={false}
                                        hasBackdrop={false}
                                        isOpen={false}
                                        lazy={true}
                                        onClose={[Function]}
                                        transitionDuration={100}
                                        transitionName="bp3-popover"
                                        usePortal={true}
                                      />
                                    </span>
                                  </Manager>
                                </Blueprint3.Popover>
                              </Blueprint3.Tooltip>
                              <Blueprint3.Text
                                ellipsize={false}
                                tagName="div"
                              >
                                <div
                                  className=""
                                >
                                  :
                                   
                                  <span />
                                </div>
                              </Blueprint3.Text>
                              <Blueprint3.Text
                                ellipsize={false}
                                tagName="div"
                              >
                                <div
                                  className=""
                                >
                                  :
                                   
                                  <span />
                                </div>
                              </Blueprint3.Text>
                              <div
                                style={
                                  Object {
                                    "margin": "0 auto",
                                    "width": "200px",
                                  }
                                }
                              >
                                <Blueprint3.Text
                                  className="bp3-text-muted"
                                  ellipsize={true}
                                  tagName="div"
                                >
                                  <div
                                    className="bp3-text-overflow-ellipsis bp3-text-muted"
                                  >
                                    :
                                     
                                    <span />
                                  </div>
                                </Blueprint3.Text>
                              </div>
                            </div>
                          </Col>
                        </div>
                      </Row>
                    </div>
                  </Blueprint3.Callout>
                </DeviceInfoCallout>
              </div>
            </Col>
            <Col
              xs={5}
            >
              <div
                className="col-xs-5"
              >
                <SharingSourcePreviewCard>
                  <Blueprint3.Card
                    className="preview-share-thumb-container"
                    elevation={0}
                    interactive={false}
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    style={
                      Object {
                        "backgroundColor": "rgba(0,0,0,0.0)",
                        "height": "200px",
                        "minWidth": "250px",
                      }
                    }
                  >
                    <div
                      className="bp3-card bp3-elevation-0 preview-share-thumb-container"
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      style={
                        Object {
                          "backgroundColor": "rgba(0,0,0,0.0)",
                          "height": "200px",
                          "minWidth": "250px",
                        }
                      }
                    >
                      <Row
                        center="xs"
                        middle="xs"
                        style={
                          Object {
                            "height": "95%",
                            "minWidth": "200px",
                          }
                        }
                      >
                        <div
                          className="row center-xs middle-xs"
                          style={
                            Object {
                              "height": "95%",
                              "minWidth": "200px",
                            }
                          }
                        >
                          <Col
                            xs={12}
                          >
                            <div
                              className="col-xs-12"
                            >
                              <Blueprint3.Spinner
                                size={60}
                              >
                                <div
                                  className="bp3-spinner"
                                >
                                  <div
                                    className="bp3-spinner-animation"
                                  >
                                    <svg
                                      height={60}
                                      strokeWidth="6.67"
                                      viewBox="1.67 1.67 96.67 96.67"
                                      width={60}
                                    >
                                      <path
                                        className="bp3-spinner-track"
                                        d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                                      />
                                      <path
                                        className="bp3-spinner-head"
                                        d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                                        pathLength={280}
                                        strokeDasharray="280 280"
                                        strokeDashoffset={210}
                                      />
                                    </svg>
                                  </div>
                                </div>
                              </Blueprint3.Spinner>
                            </div>
                          </Col>
                        </div>
                      </Row>
                      <Row
                        center="xs"
                      >
                        <div
                          className="row center-xs"
                        >
                          <Col
                            style={
                              Object {
                                "backgroundColor": "rgba(0,0,0,0.45)",
                                "color": "white",
                                "textAlign": "center",
                              }
                            }
                            xs={12}
                          >
                            <div
                              className="col-xs-12"
                              style={
                                Object {
                                  "backgroundColor": "rgba(0,0,0,0.45)",
                                  "color": "white",
                                  "textAlign": "center",
                                }
                              }
                            >
                              <Blueprint3.Text
                                ellipsize={true}
                                tagName="div"
                              >
                                <div
                                  className="bp3-text-overflow-ellipsis"
                                />
                              </Blueprint3.Text>
                            </div>
                          </Col>
                        </div>
                      </Row>
                    </div>
                  </Blueprint3.Card>
                </SharingSourcePreviewCard>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </ConfirmStep>
  </Router>
</BrowserRouter>
`;
